core/arch/aarchxx/mangle.c

```diff
@@ -914,8 +914,7 @@ insert_reachable_cti(dcontext_t *dcontext, instrlist_t *ilist, instr_t *where,
 {
     ASSERT(scratch != REG_NULL); /* required */
     /* load target into scratch register */
-    insert_mov_immed_ptrsz(dcontext, (ptr_int_t)
-                           PC_AS_JMP_TGT(dr_get_isa_mode(dcontext), target),
```

Comment:
What I'm puzzling over is the reason that 850c10c0b8c3390608715074f0ce94379d039f2f put this here: it is called out in the commit message so there must have been something that did not have the lsb set.  Not everything does: e.g., fragment tags deliberately do not have lsb set.  There must have been something like that that would make its way here.  Does a test suite run on arm hit any discrepancy here?

Reply:
I think for clean calls, we can rely on the ARM/Thumb bit being set properly in the address of the function. But I am not sure if that is true for the code DynamoRIO emits itself? Then we would need to set it explicitly when moving the address to a register.